internal/dispatch/remote/cluster_test.go

```diff
@@ -242,6 +259,228 @@ func TestSecondaryDispatch(t *testing.T) {
 	}
 }
 
+func TestLRSecondaryDispatch(t *testing.T) {
+	for _, tc := range []struct {
+		name                  string
+		expr                  string
+		request               *v1.DispatchLookupResources2Request
+		expectedDispatchCount uint32
+		expectedError         bool
+	}{
+		{
+			"no multidispatch",
+			"['invalid']",
+			&v1.DispatchLookupResources2Request{
+				ResourceRelation: &corev1.RelationReference{
+					Namespace: "somenamespace",
+					Relation:  "somerelation",
+				},
+				SubjectRelation: &corev1.RelationReference{
+					Namespace: "somenamespace",
+					Relation:  "somerelation",
+				},
+				SubjectIds: []string{"foo"},
+				TerminalSubject: &corev1.ObjectAndRelation{
+					Namespace: "foo",
+					ObjectId:  "bar",
+					Relation:  "...",
+				},
+				Metadata: &v1.ResolverMeta{DepthRemaining: 50},
+			},
+			1,
+			false,
+		},
+		{
+			"valid multidispatch",
+			"['secondary']",
+			&v1.DispatchLookupResources2Request{
+				ResourceRelation: &corev1.RelationReference{
+					Namespace: "somenamespace",
+					Relation:  "somerelation",
+				},
+				SubjectRelation: &corev1.RelationReference{
+					Namespace: "somenamespace",
+					Relation:  "somerelation",
+				},
+				SubjectIds: []string{"foo"},
+				TerminalSubject: &corev1.ObjectAndRelation{
+					Namespace: "foo",
+					ObjectId:  "bar",
+					Relation:  "...",
+				},
+				Metadata: &v1.ResolverMeta{DepthRemaining: 50},
+			},
+			2,
+			false,
+		},
+		{
+			"cursored multidispatch to invalid secondary",
+			"['secondary']",
+			&v1.DispatchLookupResources2Request{
+				ResourceRelation: &corev1.RelationReference{
+					Namespace: "somenamespace",
+					Relation:  "somerelation",
+				},
+				SubjectRelation: &corev1.RelationReference{
+					Namespace: "somenamespace",
+					Relation:  "somerelation",
+				},
+				SubjectIds: []string{"foo"},
+				TerminalSubject: &corev1.ObjectAndRelation{
+					Namespace: "foo",
+					ObjectId:  "bar",
+					Relation:  "...",
+				},
+				OptionalCursor: &v1.Cursor{
+					Sections:        []string{"somethingelse"},
+					DispatchVersion: 1,
+				},
+				Metadata: &v1.ResolverMeta{DepthRemaining: 50},
+			},
+			2, // Falls back to the default secondary.
+			false,
+		},
+		{
+			"cursored multidispatch to cursored secondary",
+			"['secondary']",
+			&v1.DispatchLookupResources2Request{
+				ResourceRelation: &corev1.RelationReference{
+					Namespace: "somenamespace",
+					Relation:  "somerelation",
+				},
+				SubjectRelation: &corev1.RelationReference{
+					Namespace: "somenamespace",
+					Relation:  "somerelation",
+				},
+				SubjectIds: []string{"foo"},
+				TerminalSubject: &corev1.ObjectAndRelation{
+					Namespace: "foo",
+					ObjectId:  "bar",
+					Relation:  "...",
+				},
+				OptionalCursor: &v1.Cursor{
+					Sections:        []string{secondaryCursorPrefix + "tertiary"},
```

Comment:
add one extra test case where the expression is `"['secondary']"` and the cursor is also for `secondary`?